File: lecture2/pset2/caesar.c
#include <stdio.h>
#include <cs50.h>
#include <string.h>
#include <ctype.h>
#include <stdlib.h>

string caesar(string word, int k);

int main(int argc, string argv[]){
    int k;
    //error
    if(argc != 2){
        printf("Usage: ./caesar k\n");
        return 1;
    }else{
        k = atoi(argv[1]);
    }

    //get word
    string word1 = get_string("plaintext:  ");
    //process
    string word2 = caesar(word1, k);
    printf("ciphertext: %s\n", word2);

    return 0;
}

string caesar(string word, int k){
    int n = strlen(word);
    for(int i=0; i<n; i++){
        if(islower(word[i])){
            word[i] = 'a' + (word[i] - 'a' + k) % 26;
        }else if(isupper(word[i])){
            word[i] = 'A' + (word[i] - 'A' + k) % 26;
        }
    }

    return word;
}

File: lecture9/pset6/caesar.py
import sys
from cs50 import get_string

def caesar(word, k):
    n = len(word)
    result = ""
    for i in range(n):
        if word[i].islower():
            num = ord('a') + (ord(word[i]) - ord('a') + k) % 26
            result += chr(num)
        elif word[i].isupper():
            num = ord('A') + (ord(word[i]) - ord('A') + k) % 26
#            print(num)
            result += chr(num)
        else:
            result += word[i]
    return result

s= len(sys.argv)
k = -1
if(s != 2):
    print("Usage: ./caesar k");
else:
    k = int(sys.argv[1])

if(k > 0):
    word1 = get_string("plaintext:  ")
    #print(word1)
    word2 = caesar(word1, k)
    print(f"ciphertext: {word2}")